esx_service/utils/auth_api.py

```diff
@@ -299,8 +299,20 @@ def _tenant_create(name, description="", vm_list=None, privileges=None):
     if not description:
         description = ""
 
-    logging.debug("_tenant_create: vms=%s", vms)
+    error_info, tenant_list = get_tenant_list_from_db()
+
+    if error_info:
+        return error_info, None
+
     vms_uuid_list = [(vm_id) for (vm_id, vm_name) in vms]
+
+    if vms_uuid_list and any_vm_part_of_another_tenant(vms_uuid_list, tenant_list):
+        error_info = error_code.generate_error_info(ErrorCode.VM_IN_ANOTHER_TENANT, vm_list, name)
+        logging.error("_tenant_create: %s", error_info.msg)
+        return error_info, None
+
+    logging.debug("_tenant_create: vms=%s", vms)
+
```

Comment:
the new code is cut-n-pasted twice. Please change it to use a function, an you wont even need a separate one for `any_vm_part_of_another_tenant` since it will be wrapped up there.